lib/array.js

```diff
@@ -201,6 +201,53 @@ const reduce = (items, fn, done, initial) => {
   fn(previous, current, next, count, items);
 };
 
+// Asynchronous reduceRight
+//   items - <Array>, incoming
+//   fn - <Function>, to be executed for each value in array
+//     previous - <any>, value previously returned in the last iteration
+//     current - <any>, current element being processed in the array
+//     callback - <Function>, callback for returning value
+//         back to reduce function
+//     counter - <number>, index of the current element
+//         being processed in array
+//     items - <Array>, the array reduce was called upon
+//   done - <Function>, on done, optional
+//     err - <Error> | <null>
+//     result - <Array>
+//   initial - <any>, optional value to be used as first
+//       argument in first iteration
+const reduceRight = (items, fn, done, initial) => {
+  done = done || common.emptyness;
+  const len = items.length;
+  let count = typeof initial === 'undefined' ? len - 2 : len - 1;
+
+  if (!len) {
+    const err = count === len - 2 ? new TypeError(EMPTY_ARR) : null;
```

Comment:
I think we need a different error message for `reduceRight`: `'Metasync: reduceRight of the empty array with no initial value'` though I don't have strong opinion about this.